inspector/src/components/globalState.ts

```diff
@@ -23,6 +24,8 @@ export class GlobalState {
     public onNewSceneObservable = new Observable<Scene>();
     public sceneImportDefaults: { [key: string]: any } = {};
 
+    public onPopupRenderObservable = new Observable<Nullable<() => ReactNode>>();

```

Comment:
Why is the type `Nullable`? Is there an architectural reason for that?

Reply:
Yes, if the user calls the observable with a null function, it is saying they don't want to render anything anymore. That's why the PopupHost checks if renderFunction exists to render the popup. An alternative could be to have two separate observables, `onPopupStartRenderObservable` and `onPopupStopRenderObservable`.

Reply:
I see. This is a little confusing then. I would expect a function that is called `onPopupRenderObservable` to notify me when the popup is about to be rendered. It should be triggered in code, and not notified by the developer directly.

A better variant would be to have a `showPersistentPopup(options)` and `hidePersistentPopup()` functions. They in turn will notify those observing this observer.